lib/run/export-file.js

```diff
@@ -77,9 +82,25 @@ module.exports = function (options, done) {
                     `error creating path for file "${path.unparsed}" for ${options.name || 'unknown-source'}`));
             }
 
-            fs.writeFile(path.unparsed, content, function (err) {
-                done(_.set(err, 'help',
-                    `error writing file "${path.unparsed}" for ${options.name || 'unknown-source'}`), path);
+            fs.stat(path.unparsed, function (error, stat) {
```

Comment:
Should we check these on start?

Reply:
As in, before the `mkirp` call, right?

Reply:
@shamasis I tried that, it ends up looking a bit repetitive:
```
        fs.stat(path.unparsed, function (error, stat) {
            if (error) {
                return mkdirp(path.dir, function (err) {
                    if (err) {
                        return done(_.set(err, 'help',
                            `error creating path for file "${path.unparsed}" for ${options.name || 'unknown-source'}`));
                    }

                    fs.writeFile(path.unparsed, content, function (err) {
                        done(_.set(err, 'help',
                            `error writing file "${path.unparsed}" for ${options.name || 'unknown-source'}`), path);
                    });
                });
            }

            var target;

            // handle cases where the specified export path is a pre-existing directory
            if (stat && stat.isDirectory()) {
                target = parsePath(options.default);

                // append timestamp
                // @todo make -0 become incremental if file name exists
                target.name += '-' + timestamp() + '0';
                target.base = target.name + target.ext;

                path.unparsed = joinPath(path.unparsed, target.base);
            }

            fs.writeFile(path.unparsed, content, function (err) {
                done(_.set(err, 'help',
                    `error writing file "${path.unparsed}" for ${options.name || 'unknown-source'}`), path);
            });
        });
```